cpp/open3d/geometry/TriangleMesh.cpp

```diff
@@ -126,7 +126,9 @@ TriangleMesh &TriangleMesh::ComputeTriangleNormals(
 
 TriangleMesh &TriangleMesh::ComputeVertexNormals(bool normalized /* = true*/) {
     ComputeTriangleNormals(false);
-    vertex_normals_.resize(vertices_.size(), Eigen::Vector3d::Zero());
+    vertex_normals_.resize(vertices_.size());
+    std::fill(vertex_normals_.begin(), vertex_normals_.end(),
+              Eigen::Vector3d::Zero());
```

Comment:
Good catch for this subtle bug: `resize(size, val)` only initializes additional copies to val and will not change existing data.